contracts/TokenLocking.sol

```diff
@@ -91,16 +91,15 @@ contract TokenLocking is TokenLockingStorage, DSMath {
   // This function is only used in context of reputation mining
   // TODO: After we add formula to calculate user's loss, refactor accordingly and/or
   // move some of the functionality to `ColonyNetworkMining` if needed
-  function punishStakers(address[] _users) public onlyReputationMiningCycle {
+  function punishStakers(address[] _stakers, address _beneficiary) public onlyReputationMiningCycle {
     address clnyToken = IColony(IColonyNetwork(colonyNetwork).getMetaColony()).getToken();
+    uint256 lostStake;
     // Passing an array so that we don't incur the EtherRouter overhead for each staker if we looped over
     // it in ReputationMiningCycle.invalidateHash;
-    for (uint256 i = 0; i < _users.length; i++) {
-      // This is pretty harsh! Are we happy with this?
-      // Alternative: lose more than they would have gained for backing the right hash.
-      userLocks[clnyToken][_users[i]].balance = 0;
-      // TODO: Where do these staked tokens go? Maybe split between the person who did the 'invalidate' transaction
-      // and the colony network?
+    for (uint256 i = 0; i < _stakers.length; i++) {
+      lostStake = userLocks[clnyToken][_stakers[i]].balance;
+      userLocks[clnyToken][_stakers[i]].balance = 0;
+      userLocks[clnyToken][_beneficiary].balance = add(userLocks[clnyToken][_beneficiary].balance, lostStake);
```

Comment:
There is no test for this functionality

Reply:
Here: https://github.com/JoinColony/colonyNetwork/pull/397/files#diff-15aaaaa81a55d904a938772d3e2a3a82R727